Allow configuration of podManagementPolicy

BigGold1310 opened the pull request:
#### What this PR does
This PR allows the configuration of the podManagementPolicy for the store-gateway statefulset through the helm chart.

#### Which issue(s) this PR fixes or relates to

Fixes #5733

#### Checklist

- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

dimitarvdimitrov commented on operations/helm/charts/mimir-distributed/values.yaml:
```diff
@@ -1366,6 +1366,9 @@ store_gateway:
   # Pod Annotations
   podAnnotations: {}
 
+  # -- Management policy for chunks-cache pods
```
```suggestion
  # -- Management policy for store-gateway pods
```

dimitarvdimitrov reviewed (commented):
awesome, thank you! I only left a minor suggestion about the comment on the value

BigGold1310 commented:
I intend to change the podManagementPolicy.  It's an alignment with other stateful sets, such as the ingester. Another aspect to consider is the improvement during zone scale-up and the recovery of a lost zone.

Ingester podManagementPolicy: https://github.com/grafana/mimir/blob/8bb9493d66693ce9102e08414f00b7785dfa2c17/operations/helm/charts/mimir-distributed/values.yaml#L808

Use-Case 1:
A few days ago, we experienced the loss of an entire zone. After the worker was recovered, Mimir began to scale up again, where the ingester simultaneously scaled the 12 replicas. However, the store-gateway scaled sequentially, taking hours to become operational again.

Use-Case two:
The documentation suggests manually "killing" an entire zone to expedite the update and rollout, where parallel scaling up would be logical. In this context, I will review the settings you mentioned.

dimitarvdimitrov commented:
> Use-Case 1:

I see. Thanks for explaining. In this case your changes make sense. However, `podManagementPolicy` is immutable on an existing StatefulSet. I suggest still adding the Helm value, but keeping the default value as `OrderedReady`. To migrate to `Parallel` you'd have to delete with `--cascade=orphan` and recreate the statefulSet with `podManagementPolicy: Parallel`

BigGold1310 commented:
@dimitarvdimitrov Agree, the chart should be backwards compatible, at least until the next version bump i.e. to 6.0.0.
From my point of view, we can set it to `OrderedReady`. During the next version change of the chart it would make sense to change it to `Parallel` to have it aligned with the other components so the same behaviour occurs for user.

How about changing it to `OrderedReady` with a comment/recommendation in the value file to move to `Parallel?

dimitarvdimitrov commented:
> During the next version change of the chart it would make sense to change it to Parallel to have it aligned with the other components so the same behaviour occurs for user.

I checked and even our jsonnet deployment uses Parallel by default. It would make sense to align helm and jsonnet. I've added this change to the [6.0 breaking changes list](https://github.com/grafana/mimir/issues/4560)

https://github.com/grafana/mimir/blob/98a9cd8cf49e2685ea86e8aaec3842d3e66ccd72/operations/mimir/common.libsonnet#L106

 
> How about changing it to `OrderedReady` with a comment/recommendation in the value file to move to `Parallel?

That's a good idea.

BigGold1310 commented:
So, I updated the PR to include `OrderedReady` as current default value with a note to switch to Parallel as it will be a change in 6.0.0. Tests are also updated to check for `OrderedReady`value.

dimitarvdimitrov reviewed (approved):
LGTM, thank you for the contribution!
